### PR TITLE
HPCC-26596 Fix problems using a source file and --main a.b@repo

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: ecl/eclcc/eclcc.cpp [hunk 1]
```diff
@@ -1828,15 +1828,15 @@ void EclCC::processFile(EclCompileInstance & instance)
         if (attributePackage)
         {
             //If attribute package is specified, resolve that package as the source for the query, and pass null to processSingleQuery
-            instance.dataServer.setown(localRepositoryManager.resolveDependentRepository(nullptr, attributePackage, false));
+            instance.dataServer.set(localRepositoryManager.queryDependentRepository(nullptr, attributePackage, false));
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp:1831]: this caused the core.

File: ecl/eclcc/eclcc.cpp [hunk 2]
```diff
@@ -1828,15 +1828,15 @@ void EclCC::processFile(EclCompileInstance & instance)
...
         }
         else
         {
             localRepositoryManager.addQuerySourceFileEclRepository(&instance.queryErrorProcessor(), querySearchPath.str(), ESFdependencies, 0);
 
             instance.dataServer.setown(localRepositoryManager.createPackage(nullptr));
         }
-        localRepositoryManager.kill();  // help ensure non-shared repositories are freed as soon as possible
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp:1839]: this caused the compile to fail - it removed the standard library.

**Review by ghalliday — COMMENTED** [review]
Other change was a rename to  avoid the set/setown confusion again.

File: ecl/eclcc/eclcc.cpp [hunk 3]
```diff
@@ -2809,7 +2809,7 @@ int EclCC::parseCommandLineOptions(int argc, const char* argv[])
         else if (iter.matchOption(optMonitorInterval, "--monitorinterval"))
         {
         }
-        else if (iter.matchOption(tempArg, "-main"))
+        else if (iter.matchOption(tempArg, "-main") || iter.matchOption(tempArg, "--main"))
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp:2812]: for compatibility with ecl run.  I will add a jira to make all options compatible.

**Review by richardkchapman — APPROVED** [review]

File: ecl/hql/hqlrepository.cpp [hunk 4]
```diff
@@ -584,7 +584,7 @@ void EclRepositoryManager::processArchive(IPropertyTree * archiveTree)
     addRepository(archiveCollection, nullptr, true);
 }
 
-IEclPackage * EclRepositoryManager::resolveDependentRepository(IIdAtom * name, const char * defaultUrl, bool requireSHA)
+IEclPackage * EclRepositoryManager::queryDependentRepository(IIdAtom * name, const char * defaultUrl, bool requireSHA)
```
Comment (shamser) [inline ecl/hql/hqlrepository.cpp:587]: I'm not sure that I understand why this function has been changed to a query* function.  I though query* functions should be returning anything that needs to be freed by the caller.  Please can you clarify.

Comment (ghalliday) [reply ecl/hql/hqlrepository.cpp:587]: no, query() functions do not need freeing by the caller.  get() functions do need to be freeing.

**Review by shamser — COMMENTED** [review]
I'm not sure that I understand why this function has been changed to a query* function. I though query* functions should be returning anything that needs to be freed by the caller. Please can you clarify.

**Review by shamser — APPROVED** [review]

**Pull request merged**